src/story/article.py

```diff
@@ -12,52 +12,50 @@
 from src.data_models.article_model import ArticleModel
 
 parser = reqparse.RequestParser()
-parser.add_argument("story_id")
+parser.add_argument("article_id")
 parser.add_argument("user_id")
 
 
-def get_articles(user_id, article_id=None):
+def get_articles(filter):
     """Get articles filtered by article_id, user_id if available.
 
     Send request to SQL database to get articles based on article_id,
-    user_id. If no article_id is parsed, we return all articles
+    user_id and other filtering keys. If no article_id is parsed, we return all articles
 
     Args:
-        article_id: int, unique article id
-        user_id: int, unique user id
+        filter: dict, defining filtering scheme
     Return:
         A list of article dictionary.
     """
-    args = {}
-    if article_id:
-        args["article_id"] = article_id
-    if user_id:
-        args["user_id"] = user_id
-    articles = ArticleModel.query.filter_by(**args).all()
+    column_keys = ArticleModel.__table__.columns.keys()
+    for filter_key in filter.keys():
+        if filter_key not in column_keys:
```

Comment:
if the filter key is not in column keys, it is better to return an empty response result.

Reply:
Or shall we return 400 invalid parameters?